analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/ProgramState.cs

```diff
@@ -79,6 +79,9 @@ private ProgramState()
                 ? this with { OperationValue = OperationValue.Remove(ResolveCapture(operation)) }
                 : this with { OperationValue = OperationValue.SetItem(ResolveCapture(operation), value) };
 
+        public ProgramState SetOperationConstraint(IOperationWrapper operation, SymbolicConstraint constraint) =>
+            SetOperationValue(operation, (this[operation] ?? SymbolicValue.Empty).WithConstraint(constraint));
```

Comment:
This should look the same as the overload for `IOperationWrapperSonar` - only call the real canonical implementation, without duplicating the logic. It's basically just parameter unwrapping forward